core/common/assets/js/api/core/data.js

```diff
@@ -0,0 +1,476 @@
+	DELETABLE = [ 'DELETE' ],
+	ALLMETHODS = [ 'GET', 'POST', 'PUT', 'PATCH', 'DELETE' ];
+
+export default class Data extends Commands {
+	constructor( args = {} ) {
+		super( args );
+
+		this.args = Object.assign( args, {
+			namespace: 'elementor',
+			version: '1',
+		} );
+
+		this.cache = new Cache( this );
+		this.validatedRequests = {};
+
+		this.baseEndpointAddress = '';
+
+		elementorCommon.elements.$window.on( 'elementor:loaded', this.onElementorLoaded.bind( this ) );
+	}
+
+	onElementorLoaded() {
+		const { namespace, version } = this.args;
+
+		this.baseEndpointAddress = `${ elementorCommon.config.rest_url }${ namespace }/v${ version }/`;
+	}
+
+	/**
+	 * Function getHTTPMethod().
+	 *
+	 * Returns HTTP Method by type.
+	 *
+	 * @param {DataTypes} type
+	 *
+	 * @return {string|boolean}
+	 */
+	getHTTPMethod( type ) {
+		switch ( type ) {
+			case 'create':
+				return 'POST';
+
+			case 'delete':
+				return 'DELETE';
+
+			case 'get':
+				return 'GET';
+
+			case 'update':
+				return 'PUT';
+		}
+
+		return false;
+	}
+
+	/**
+	 * Function getAllowedMethods().
+	 *
+	 * Returns allowed HTTP methods by type.
+	 *
+	 * @param {DataTypes} type
+	 *
+	 * @return {[string]|boolean}
+	 */
+	getAllowedMethods( type ) {
+		switch ( type ) {
+			case 'create':
+				return CREATABLE;
+
+			case 'delete':
+				return DELETABLE;
+
+			case 'get':
+				return READABLE;
+
+			case 'update':
+				return EDITABLE;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Function commandToEndpoint().
+	 *
+	 * Convert command to endpoint.
+	 *
+	 * For example `component/command/:arg_example` => `controller/endpoint/8`.
+	 *
+	 * TODO: Find a better solution.
+	 *
+	 * @param {string} command
+	 * @param {{}} args
+	 * @param {string|null} [format]
+	 *
+	 * @returns {string} endpoint
+	 */
+	commandToEndpoint( command, args, format = null ) {
+		let endpoint = command;
+
+		if ( args.query ) {
+			if ( ! format ) {
+				format = command;
+			}
+
+			if ( format && format.includes( '/{' ) ) {
+				// Means command includes magic query arguments ( controller/endpoint/{whatever} ).
+				const magicParams = format.split( '/' ).filter( ( str ) => '{' === str.charAt( 0 ) );
+
+				magicParams.forEach( ( param ) => {
+					// Remove the '{', '}'.
+					param = param.replace( '{', '' );
+					param = param.replace( '}', '' );
+
+					const formatted = Object.entries( args.query ).find( ( [ key ] ) => key === param );
+
+					if ( ! formatted ) {
+						return;
+					}
+
+					const key = formatted[ 0 ],
+						value = formatted[ 1 ].toString();
+
+					// Replace magic params with values.
+					format = format.replace( new RegExp( '{' + param + '}', 'g' ), value );
+
+					delete args.query[ key ];
+				} );
+
+				endpoint = format;
+			}
+
+			if ( endpoint.includes( 'index' ) ) {
+				endpoint = endpoint.replace( '/index', '' );
+			}
+
+			if ( args.query.id ) {
+				endpoint += '/' + args.query.id.toString();
+				delete args.query.id;
+			}
```

Comment:
It should be in the format.

Reply:
Handled: https://github.com/elementor/elementor/pull/11388/commits/cd9bc3da5a29fdf323158b3c81bc47e49d1c40e1